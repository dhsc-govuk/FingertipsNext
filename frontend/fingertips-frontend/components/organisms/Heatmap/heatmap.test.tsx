import { render } from '@testing-library/react';
import { Heatmap } from '.';
import {
  placeholderGroupAreaCode,
  placeholderHeatmapIndicatorData as placeholderIndicatorData,
} from './heatmapUtil.test';
import {
  areaCodeForEngland,
  englandAreaString,
} from '@/lib/chartHelpers/constants';

it('snapshot test - england benchmark', () => {
  const container = render(
    <Heatmap
      indicatorData={placeholderIndicatorData}
      groupAreaCode={placeholderGroupAreaCode}
      benchmarkAreaCode={areaCodeForEngland}
      benchmarkAreaName={englandAreaString}
    />
  );

  expect(container.asFragment()).toMatchSnapshot();
});

it('snapshot test - group area benchmark', () => {
  const container = render(
    <Heatmap
      indicatorData={placeholderIndicatorData}
      groupAreaCode={placeholderGroupAreaCode}
      benchmarkAreaCode={placeholderGroupAreaCode}
      benchmarkAreaName={
        placeholderIndicatorData[0].healthDataForAreas.find((healthData) => {
          return healthData.areaCode === placeholderGroupAreaCode;
        })?.areaName ?? 'no group area in data?'
      }
    />
  );

  expect(container.asFragment()).toMatchSnapshot();
});
