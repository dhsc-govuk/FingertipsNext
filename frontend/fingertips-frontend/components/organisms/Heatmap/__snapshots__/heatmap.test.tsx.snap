// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  .c4 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 14px;
  line-height: 1;
  display: inline-block;
  padding-top: 5px;
  padding-right: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  outline: 2px solid transparent;
  outline-offset: -2px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  background-color: #1d70b8;
  color: #ffffff;
}

.c5 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #00703C;
  tint: SOLID;
}

.c6 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #FFDD00;
  color: #0B0C0C;
}

.c7 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #D4351C;
  tint: SOLID;
}

.c8 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: transparent;
  color: #0B0C0C;
  border: 1px solid #0B0C0C;
}

.c9 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #5694CA;
}

.c10 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #003078;
  tint: SOLID;
}

.c11 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #85994B;
  color: #0B0C0C;
}

.c12 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #831504;
  tint: SOLID;
}

.c13 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #AFCFE5;
  color: #0B0C0C;
}

.c14 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #1D70B8;
  tint: SOLID;
}

.c15 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #E4DDFF;
  color: #0B0C0C;
}

.c16 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #CBBEF4;
  color: #0B0C0C;
}

.c17 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #AA90EC;
  color: #0B0C0C;
}

.c18 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #8B60E2;
  tint: SOLID;
}

.c19 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #6B33C3;
  tint: SOLID;
}

.c20 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #D494C1;
  color: #0B0C0C;
}

.c21 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #BC6AAA;
  color: #0B0C0C;
}

.c22 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #A44596;
  tint: SOLID;
}

.c23 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #812972;
  tint: SOLID;
}

.c24 {
  padding: 5px 8px 4px 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.3125em;
  font-weight: 300;
  text-transform: unset;
  background-color: #561950;
  tint: SOLID;
}

.c3 {
  position: relative;
  left: -5px;
}

.c2 {
  font-family: nta,Arial,sans-serif;
  font-weight: 300;
  font-size: 0.842em;
}

.c0 {
  margin-bottom: 2em;
}

.c1 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  margin: 1em 0.1em 0.1em 0;
  font-family: nta,Arial,sans-serif;
  font-weight: 300;
}

@media print {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {
  .c4 {
    font-size: 14px;
    line-height: 1;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    font-size: 16px;
    line-height: 1;
  }
}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

<div
    class="c0"
    data-testid="benchmarkLegend-component"
  >
    <h4
      class="c1"
    >
      Compared to England
    </h4>
    <span
      class="c2"
    >
      95% confidence
    </span>
    <div
      class="c3"
    >
      <strong
        class="c4 c5"
      >
        Better
      </strong>
      <strong
        class="c4 c6"
      >
        Similar
      </strong>
      <strong
        class="c4 c7"
      >
        Worse
      </strong>
      <strong
        class="c4 c8"
      >
        Not compared
      </strong>
      <strong
        class="c4 c9"
      >
        Lower
      </strong>
      <strong
        class="c4 c10"
      >
        Higher
      </strong>
    </div>
    <span
      class="c2"
    >
      99.8% confidence
    </span>
    <div
      class="c3"
    >
      <strong
        class="c4 c11"
      >
        Better
      </strong>
      <strong
        class="c4 c6"
      >
        Similar
      </strong>
      <strong
        class="c4 c12"
      >
        Worse
      </strong>
      <strong
        class="c4 c8"
      >
        Not compared
      </strong>
      <strong
        class="c4 c13"
      >
        Lower
      </strong>
      <strong
        class="c4 c14"
      >
        Higher
      </strong>
    </div>
    <h4
      class="c1"
    >
      Quintiles
    </h4>
    <div
      class="c3"
    >
      <strong
        class="c4 c15"
      >
        Lowest
      </strong>
      <strong
        class="c4 c16"
      >
        Low
      </strong>
      <strong
        class="c4 c17"
      >
        Middle
      </strong>
      <strong
        class="c4 c18"
      >
        High
      </strong>
      <strong
        class="c4 c19"
      >
        Highest
      </strong>
    </div>
    <div
      class="c3"
    >
      <strong
        class="c4 c20"
      >
        Worst
      </strong>
      <strong
        class="c4 c21"
      >
        Worse
      </strong>
      <strong
        class="c4 c22"
      >
        Middle
      </strong>
      <strong
        class="c4 c23"
      >
        Better
      </strong>
      <strong
        class="c4 c24"
      >
        Best
      </strong>
    </div>
  </div>
  .c5 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  margin-top: 0;
  margin-bottom: 15px;
}

.c3 {
  padding: 10px 20px 10px 0;
  border-bottom: 1px solid #bfc1c3;
  text-align: left;
  font-weight: 700;
}

.c3:last-child {
  padding-right: 0;
}

.c14 {
  padding: 10px 20px 10px 0;
  border-bottom: 1px solid #bfc1c3;
  text-align: left;
}

.c14:last-child {
  padding-right: 0;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
  margin-bottom: 20px;
}

.c10 {
  -webkit-transform: translate(-15px) rotate(30deg);
  -ms-transform: translate(-15px) rotate(30deg);
  transform: translate(-15px) rotate(30deg);
  -webkit-transform-origin: bottom right;
  -ms-transform-origin: bottom right;
  transform-origin: bottom right;
}

.c11 {
  -webkit-transform: scale(-1);
  -ms-transform: scale(-1);
  transform: scale(-1);
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  max-width: 40px;
  -webkit-writing-mode: vertical-lr;
  -ms-writing-mode: tb;
  writing-mode: vertical-lr;
  max-height: 300px;
  display: block;
}

.c12 {
  background-color: #B1B4B6;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 4px;
  padding-right: 4px;
}

.c13 {
  background-color: #F3F2F1;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 4px;
  padding-right: 4px;
}

.c6 {
  height: 30px;
}

.c7 {
  width: 240px;
}

.c4 {
  vertical-align: bottom;
  width: 240px;
}

.c8 {
  vertical-align: bottom;
  padding-right: 0px;
}

.c9 {
  vertical-align: bottom;
  width: 60px;
  padding-right: 0px;
}

.c16 {
  min-height: 70px;
  padding-right: 0;
}

.c18 {
  text-align: center;
  min-height: 70px;
  width: 60px;
  padding: 0;
}

.c19 {
  color: #0B0C0C;
  background-color: #B1B4B6;
}

.c21 {
  color: #0B0C0C;
  background-color: #FFFFFF;
}

.c22 {
  color: #FFFFFF;
  background-color: #6B33C3;
}

.c20 {
  min-width: 60px;
  max-width: 80px;
  display: inline;
}

.c15 {
  text-overflow: ellipsis;
  overflow: hidden;
  width: 240px;
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
}

.c17 {
  min-width: 40px;
  padding-right: 20px;
}

.c2 {
  display: block;
  width: 100%;
  table-layout: fixed;
}

.c0 {
  overflow-x: scroll;
  max-width: 960px;
}

@media print {
  .c5 {
    color: #000;
  }
}

@media print {
  .c5 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    margin-bottom: 20px;
  }
}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 30px;
  }
}

<div
    class="c0"
  >
    <table
      class="c1 c2"
      data-testid="heatmapChart-component"
    >
      <tbody
        class=""
      >
        <tr
          class=""
        >
          <th
            class="c3 c4"
          >
            <h4
              class="c5 c6 c7"
            >
              Indicators
            </h4>
          </th>
          <th
            class="c3 c8"
          >
            <h4
              class="c5 c6"
            >
              Value unit
            </h4>
          </th>
          <th
            class="c3 c8"
          >
            <h4
              class="c5 c6"
            >
              Period
            </h4>
          </th>
          <th
            class="c3 c9"
          >
            <div
              class="c10"
            >
              <h4
                class="c5 c11 c12"
              >
                Benchmark: England
              </h4>
            </div>
          </th>
          <th
            class="c3 c9"
          >
            <div
              class="c10"
            >
              <h4
                class="c5 c11 c13"
              >
                Dentdale
              </h4>
            </div>
          </th>
          <th
            class="c3 c9"
          >
            <div
              class="c10"
            >
              <h4
                class="c5 c11"
              >
                Comedically Long Area Name with a Devilishly Difficult Distance to Display
              </h4>
            </div>
          </th>
          <th
            class="c3 c9"
          >
            <div
              class="c10"
            >
              <h4
                class="c5 c11"
              >
                Garsdale
              </h4>
            </div>
          </th>
        </tr>
        <tr
          class=""
        >
          <td
            class="c14"
            data-testid="heatmap-cell-indicator-title"
          >
            <div
              class="c15"
            >
              Donkey / Goose ratio
            </div>
          </td>
          <td
            class="c14 c16"
            data-testid="heatmap-cell-indicator-info"
          >
            <div
              class="c17"
            >
              %
            </div>
          </td>
          <td
            class="c14 c16"
            data-testid="heatmap-cell-indicator-info"
          >
            <div
              class="c17"
            >
              2002
            </div>
          </td>
          <td
            class="c14 c18 c19"
            data-testid="heatmap-cell-data"
          >
            <div
              class="c20"
            >
              <div
                class="c20"
              >
                1,112.0
              </div>
            </div>
          </td>
          <td
            class="c14 c18 c21"
            data-testid="heatmap-cell-data"
          >
            <div
              class="c20"
            >
              <div
                class="c20"
              >
                1,132.0
              </div>
            </div>
          </td>
          <td
            class="c14 c18 c21"
            data-testid="heatmap-cell-data"
          >
            <div
              class="c20"
            >
              <div
                class="c20"
              >
                1,142.0
              </div>
            </div>
          </td>
          <td
            class="c14 c18 c21"
            data-testid="heatmap-cell-data"
          >
            <div
              class="c20"
            >
              <div
                class="c20"
              >
                1,122.0
              </div>
            </div>
          </td>
        </tr>
        <tr
          class=""
        >
          <td
            class="c14"
            data-testid="heatmap-cell-indicator-title"
          >
            <div
              class="c15"
            >
              Rate of walkers tripping over sheep
            </div>
          </td>
          <td
            class="c14 c16"
            data-testid="heatmap-cell-indicator-info"
          >
            <div
              class="c17"
            >
              per 100
            </div>
          </td>
          <td
            class="c14 c16"
            data-testid="heatmap-cell-indicator-info"
          >
            <div
              class="c17"
            >
              2002
            </div>
          </td>
          <td
            class="c14 c18 c19"
            data-testid="heatmap-cell-data"
          >
            <div
              class="c20"
            >
              <div
                class="c20"
              >
                112.0
              </div>
            </div>
          </td>
          <td
            class="c14 c18 c21"
            data-testid="heatmap-cell-data"
          >
            <div
              class="c20"
            >
              <div
                class="c20"
              >
                132.0
              </div>
            </div>
          </td>
          <td
            class="c14 c18 c21"
            data-testid="heatmap-cell-data"
          >
            <div
              class="c20"
            >
              <div
                class="c20"
              >
                142.0
              </div>
            </div>
          </td>
          <td
            class="c14 c18 c21"
            data-testid="heatmap-cell-data"
          >
            <div
              class="c20"
            >
              <div
                class="c20"
              >
                X
              </div>
            </div>
          </td>
        </tr>
        <tr
          class=""
        >
          <td
            class="c14"
            data-testid="heatmap-cell-indicator-title"
          >
            <div
              class="c15"
            >
              Very Verbose Indicator Name With an Extreeeeeeeme Number of Words to Try And Trip Up The View. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus varius magna massa, commodo consectetur erat hendrerit id. In semper, nibh eu efficitur sagittis, quam lectus semper augue, quis vestibulum ipsum urna ut orci.
            </div>
          </td>
          <td
            class="c14 c16"
            data-testid="heatmap-cell-indicator-info"
          >
            <div
              class="c17"
            >
              per 1000
            </div>
          </td>
          <td
            class="c14 c16"
            data-testid="heatmap-cell-indicator-info"
          >
            <div
              class="c17"
            >
              2004
            </div>
          </td>
          <td
            class="c14 c18 c19"
            data-testid="heatmap-cell-data"
          >
            <div
              class="c20"
            >
              <div
                class="c20"
              >
                X
              </div>
            </div>
          </td>
          <td
            class="c14 c18 c21"
            data-testid="heatmap-cell-data"
          >
            <div
              class="c20"
            >
              <div
                class="c20"
              >
                X
              </div>
            </div>
          </td>
          <td
            class="c14 c18 c22"
            data-testid="heatmap-cell-data"
          >
            <div
              class="c20"
            >
              <div
                class="c20"
              >
                41.0
              </div>
            </div>
          </td>
          <td
            class="c14 c18 c21"
            data-testid="heatmap-cell-data"
          >
            <div
              class="c20"
            >
              <div
                class="c20"
              >
                X
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
